doc/netplan.md

```diff
@@ -1007,6 +1008,60 @@ Examples:
 
 :   Alternate name for the ``key`` field. See above.
 
+Wireguard-specific keys:
+
+    ``private-key`` (scalar) – since **0.100**
+    :   Base64-encoded private key, prefixed with ``base64:``. Or an absolute
+        path to a file with the private key.
+
+    ``fwmark`` (scalar) – since **0.100**
+    :   Firewall mark for outgoing WireGuard packets from this interface,
+        optional.
+
+    ``port`` (scalar) – since **0.100**
+    :   UDP port to listen at or ``auto``. Optional, defaults to ``auto``.
+
+    ``peers`` (sequence of mappings) – since **0.100**
+    :   A list of peers, each having keys documented below.
+
+    Example:
+
+    tunnels:
+        wireguard0:
+            mode: wireguard
+            peers:
+                - public-key: /some/public.key
+                  allowed-ips: [0.0.0.0/0, "2001:fe:ad:de:ad:be:ef:1/24"]
+                  keepalive: 23
+                  remote: 1.2.3.4:5
+                - public-key: /other/public.key
+                  allowed-ips: [0.0.0.0/0, "10.10.10.20/24"]
+                  keepalive: 22
+                  remote: 5.4.3.2:1
+
+    ``public-key`` (scalar) – since **0.100**
+    :   Peer's base64-encoded public key, required.
+
+    ``shared-key`` (scalar) – since **0.100**
+    :   Optional base64-encoded pre shared key for a peer, prefixed with
+        ``base64:``. Or an absolute path to a file with a pre shared key.
+
+    ``remote`` (scalar) – since **0.100**
+    :   Remote endpoint IPv4/IPv6 address or a hostname, followed by a colon
+        and a port number.
+
+    ``allowed-ips`` (sequence of scalars) – since **0.100**
+    :    A list of IP (v4 or v6) addresses with CIDR masks from which this peer
+         is allowed to send incoming traffic and to which outgoing traffic for
+         this peer is directed. The catch-all 0.0.0.0/0 may be specified for
+         matching all IPv4 addresses, and ::/0 may be specified for matching
+         all IPv6 addresses.
```

Comment:
I wonder if this should actually be expressed as 'routes' on the tunnel, possibly also reusing the 'use-routes: false' syntax from the dhcp*-overrides.  Does the backend implement this via managing the routing table?

Reply:
Indeed.. there is quite some overlap of the `routes` and `peers` stanza. For Wireguard the backends do **not** implement it via routing tables, tho. There are explicit settings for both backends, to set "AllowedIPs" and "Endpoints".

Lets see if I can implement `routes` for Wireguard tunnels instead of introducing this `peers` setting.

Reply:
`peers:` is now merged with `routes:`, see comment above.